drivers/ipm/ipm_imx.c

```diff
@@ -175,7 +175,7 @@ static int imx_mu_ipm_send(const struct device *dev, int wait, uint32_t id,
 		return -EINVAL;
 	}
 
-	if (size > CONFIG_IPM_IMX_MAX_DATA_SIZE) {
+	if ((size < 0) || (size > CONFIG_IPM_IMX_MAX_DATA_SIZE)) {
```

Comment:
I agree with this check, is valid.
Can you please check in other drivers too and update this PR - some examples:
- https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/ipm/ipm_mhu.c#L71
- https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/ipm/ipm_cavs_host.c#L59

Reply:
only cavs was having this problem, the other driver does not mixed signed/unsigned so I didn't any check there.